### ISSUE TITLE
Metrics API produces invalid metric names for Prometheus

### ISSUE THREAD

**Issue by nlippke** [payara/Payara#3240]
# Description #
----------

If you follow the [guideline](https://docs.payara.fish/documentation/microprofile/metrics.html) for setting up metrics export, you may end up with metrics which cannot be consumed by promerheus:

Using metrics.xml containing:
```
<metadata>
     <name>jdbc.connection.pool.%sPool.numconnfree</name>
     <mbean>amx:pp=/mon/server-mon[server],type=jdbc-connection-pool-mon,name=resources/%sPool/numconnfree#current</mbean>
     <type>counter</type>
     <unit>none</unit>
     <displayName>numconnfreePool</displayName>
     <description>The total number of free connections in the pool as of the last sampling.</description>
</metadata>
```
produces:
```
# TYPE vendor:jdbc_connection_pool_resources/_timer_pool_pool_numconnfree counter
# HELP vendor:jdbc_connection_pool_resources/_timer_pool_pool_numconnfree The total number of free connections in the pool as of the last sampling.
vendor:jdbc_connection_pool_resources/_timer_pool_pool_numconnfree 0
# TYPE vendor:jdbc_connection_pool_resources/_derby_pool_pool_numconnfree counter
# HELP vendor:jdbc_connection_pool_resources/_derby_pool_pool_numconnfree The total number of free connections in the pool as of the last sampling.
vendor:jdbc_connection_pool_resources/_derby_pool_pool_numconnfree 0
# TYPE vendor:jdbc_connection_pool_resources/_h2_pool_pool_numconnfree counter
# HELP vendor:jdbc_connection_pool_resources/_h2_pool_pool_numconnfree The total number of free connections in the pool as of the last sampling.
vendor:jdbc_connection_pool_resources/_h2_pool_pool_numconnfree 0
```
But slashes (/) are not allowed in metric names as described [here](https://prometheus.io/docs/instrumenting/writing_exporters/#naming).

> Only `[a-zA-Z0-9:_]` are valid in metric names, any other characters should be sanitized to an underscore.

## Expected Outcome
```
# TYPE vendor:jdbc_connection_pool_resources_timer_pool_pool_numconnfree counter
# HELP vendor:jdbc_connection_pool_resources_timer_pool_pool_numconnfree The total number of free connections in the pool as of the last sampling.
vendor:jdbc_connection_pool_resources_timer_pool_pool_numconnfree 0
# TYPE vendor:jdbc_connection_pool_resources_derby_pool_pool_numconnfree counter
# HELP vendor:jdbc_connection_pool_resources_derby_pool_pool_numconnfree The total number of free connections in the pool as of the last sampling.
vendor:jdbc_connection_pool_resources_derby_pool_pool_numconnfree 0
# TYPE vendor:jdbc_connection_pool_resources_h2_pool_pool_numconnfree counter
# HELP vendor:jdbc_connection_pool_resources_h2_pool_pool_numconnfree The total number of free connections in the pool as of the last sampling.
vendor:jdbc_connection_pool_resources_h2_pool_pool_numconnfree 0
```
## Environment ##

- **Payara Version**: 4.x, 5.x
- **Edition**: full,micro
- **JDK Version**: doesn't matter
- **Operating System**: doesn't matter
- **Database**: doesn't matter

**Comment by jGauravGupta** [comment]
Created internal ticket PAYARA-3260 to address the issue.

**Comment by jGauravGupta** [comment]
Hi @nlippke,
Thanks for reporting the issue, I have created internal ticket PAYARA-3260 to address.

**Comment by Ykus** [comment]
Hi,

pull request is ready.

Zdeněk Soukup

**Comment by smillidge** [comment]
Fixed by #3633
